folding-schemes/src/folding/hypernova/serialize.rs

```diff
@@ -0,0 +1,422 @@
+use crate::arith::ccs::CCS;
+use crate::arith::r1cs::R1CS;
+use crate::folding::hypernova::ProverParams;
+use crate::folding::hypernova::VerifierParams;
+use ark_crypto_primitives::sponge::poseidon::PoseidonConfig;
+use ark_crypto_primitives::sponge::Absorb;
+use ark_ec::{CurveGroup, Group};
+use ark_ff::PrimeField;
+use ark_r1cs_std::groups::{CurveVar, GroupOpsBounds};
+use ark_r1cs_std::ToConstraintFieldGadget;
+use ark_serialize::CanonicalDeserialize;
+use ark_serialize::{CanonicalSerialize, Compress, SerializationError, Validate};
+use ark_std::marker::PhantomData;
+
+use crate::folding::hypernova::cccs::CCCS;
+use crate::folding::hypernova::lcccs::LCCCS;
+use crate::folding::hypernova::Witness;
+use crate::folding::nova::{
+    CommittedInstance as CycleFoldCommittedInstance, Witness as CycleFoldWitness,
+};
+use crate::FoldingScheme;
+use crate::{
+    commitment::CommitmentScheme,
+    folding::{circuits::CF2, nova::PreprocessorParam},
+    frontend::FCircuit,
+};
+
+use super::HyperNova;
+
+impl<C1, GC1, C2, GC2, FC, CS1, CS2, const MU: usize, const NU: usize, const H: bool>
+    CanonicalSerialize for HyperNova<C1, GC1, C2, GC2, FC, CS1, CS2, MU, NU, H>
+where
+    C1: CurveGroup,
+    GC1: CurveVar<C1, CF2<C1>> + ToConstraintFieldGadget<CF2<C1>>,
+    C2: CurveGroup,
+    GC2: CurveVar<C2, CF2<C2>>,
+    FC: FCircuit<C1::ScalarField>,
+    CS1: CommitmentScheme<C1, H>,
+    CS2: CommitmentScheme<C2, H>,
+{
+    fn serialize_compressed<W: std::io::prelude::Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::Yes)
+    }
+
+    fn compressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::Yes)
+    }
+
+    fn serialize_uncompressed<W: std::io::prelude::Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::No)
+    }
+
+    fn uncompressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::No)
+    }
+
+    fn serialize_with_mode<W: std::io::prelude::Write>(
+        &self,
+        mut writer: W,
+        compress: ark_serialize::Compress,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        //self.ccs.serialize_with_mode(&mut writer, compress)?;
```

Comment:
Leftover? Not sure if I would serialize it together..

Reply:
Good catch, indeed the idea is to not serialize it.